packages/data/src/persist.js

```diff
@@ -19,63 +14,62 @@ export function setPersistenceStorage( storage ) {
  *
  * @return {Function} Enhanced reducer.
  */
-export function withRehydration( reducer, reducerKey, storageKey ) {
+export function withRehydration( reducer ) {
 	// EnhancedReducer with auto-rehydration
 	const enhancedReducer = ( state, action ) => {
-		const nextState = reducer( state, action );
-
-		if ( action.type === 'REDUX_REHYDRATE' && action.storageKey === storageKey ) {
-			return {
-				...nextState,
-				[ reducerKey ]: action.payload,
-			};
+		if ( action.type === 'REDUX_REHYDRATE' ) {
+			return reducer( action.payload, {
+				...action,
+				previousState: state,
+			} );
 		}
 
-		return nextState;
+		return reducer( state, action );
 	};
 
 	return enhancedReducer;
 }
 
 /**
- * Loads the initial state and persist on changes.
+ * Higher-order reducer used to persist just one key from the reducer state.
  *
- * This should be executed after the reducer's registration.
+ * @param {function} reducer    Reducer function.
+ * @param {string} keyToPersist The reducer key to persist.
  *
- * @param {Object}   store      Store to enhance.
- * @param {Function} reducer    The reducer function. Used to get default values and to allow custom serialization by the reducers.
- * @param {string}   reducerKey The reducer key to persist (example: reducerKey.subReducerKey).
- * @param {string}   storageKey The storage key to use.
+ * @return {function} Updated reducer.
  */
-export function loadAndPersist( store, reducer, reducerKey, storageKey ) {
-	// Load initially persisted value
-	const persistedString = persistenceStorage.getItem( storageKey );
-	if ( persistedString ) {
-		const persistedState = {
-			...get( reducer( undefined, { type: '@@gutenberg/init' } ), reducerKey ),
-			...JSON.parse( persistedString ),
-		};
+export function restrictPersistence( reducer, keyToPersist ) {
+	return ( state, action ) => {
+		const nextState = reducer( state, action );
 
-		store.dispatch( {
-			type: 'REDUX_REHYDRATE',
-			payload: persistedState,
-			storageKey,
-		} );
-	}
+		if ( action.type === 'SERIALIZE' ) {
+			return { [ keyToPersist ]: nextState[ keyToPersist ] };
+		}
 
-	// Persist updated preferences
-	let currentStateValue = get( store.getState(), reducerKey );
-	store.subscribe( () => {
-		const newStateValue = get( store.getState(), reducerKey );
-		if ( newStateValue !== currentStateValue ) {
-			currentStateValue = newStateValue;
-			const stateToSave = get( reducer( store.getState(), { type: 'SERIALIZE' } ), reducerKey );
-			persistenceStorage.setItem( storageKey, JSON.stringify( stateToSave ) );
+		if ( action.type === 'REDUX_REHYDRATE' ) {
+			return {
+				...action.previousState,
+				...state,
+				[ keyToPersist ]: {
+					...action.previousState[ keyToPersist ],
+					...state[ keyToPersist ],
+				},
+			};
 		}
-	} );
+
+		return nextState;
+	};
+}
+
+/**
+ * Get the persistence storage handler.
+ *
+ * @return {Object} Persistence storage.
+ */
+export function getPersistenceStorage() {
```

Comment:
It would be nice to have it located in the file next to the `setPersistenceStorage` method.

I also think that since we introduce this method we could do a very tiny refactor:
```js
let persistenceStorage;

export function getPersistenceStorage() {
    return persistenceStorage || window. localStorage;
}
```

to ensure it never tries to reference `window` when you set persistence when doing setup. I think at the moment it might error in the env where `window` is not present.